### PR TITLE
Perceptron

### PR DESCRIPTION
**Pull request by MarcCote**
Simple model to test the smartpy library.

### CODE REVIEW

File: perceptron/train_perceptron.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+def train_simple_perceptron():
+    with Timer("Loading dataset"):
+        trainset, validset, testset = load_dataset('mnist')
+
+    with Timer("Creating model"):
+        input_size = trainset.input_size
+        # TODO: We should the number of different targets in the dataset,
+        #       but I'm not sure how to do it right (keep in mind the regression?).
+        output_size = 10
+        model = Perceptron(input_size, output_size)
```
Comment (mgermain) [inline perceptron/train_perceptron.py]: why not :

```
model = Perceptron(trainset.input_size, output_size)
```

File: perceptron/perceptron.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        self.W.set_value(weights_initializer(self.W.get_value().shape))
+
+    @property
+    def parameters(self):
+        return {'W': self.W, 'b': self.b}
+
+    def fprop(self, X):
+        preactivation = T.dot(X, self.W) + self.b
+        probs = T.nnet.softmax(preactivation)
+        return probs
```
Comment (mgermain) [inline perceptron/perceptron.py]: Should probaly return
```
return probs.eval()
```

**Comment by MarcCote** [comment]
@ASalvail @mgermain I made a new commit fixing some issues you raised, porting the perceptron to Python 3 and removing the MLPython dependency.

**Comment by MarcCote** [comment]
@ASalvail @mgermain This is ready to be merged.

**Pull request merged**